### DISCUSSION TITLE
Automatically approved refunds on the lightning network should not need an expiration time >1 day

### DISCUSSION THREAD

**Discussion by PatMulligan** [btcpayserver/btcpayserver#3917] [General]
I tried to issue a refund to someone who uses Muun wallet. I sent them the link to the refund page and they tried to paste in their lightning invoice address, however, there is a problem in that Muun invoices have a fixed expiration time of 24 hours. On the other hand, BTCPay Server would only allow me to accept invoices with at least 1 day expiration time.

If it's a lightning refund that is set to automatically approve, then it doesn't seem to me like a >1 day invoice expiration time is necessary. As it stands, this blocks all Muun wallet users from interacting with the lightning refund functionality.

**Comment by pavlenex** [comment]
You can configure this in Store Settings > Payments.
If I am not mistaken the default is 30 days.

![Screenshot 2022-06-30 at 10 29 43](https://user-images.githubusercontent.com/36959754/176630927-abba3c7f-e725-479b-9ca8-3789d936c540.png)

**Comment by PatMulligan** [comment]
yes I tried that! In the case of Muun, they set their invoice expiration to 24 hours and the minimum value that BTCPay Server accepts it 1 day. It doesn't work out

**Reply by pavlenex** [reply]
cc @Kukks how is this possible?

**Reply by bolatovumar** [reply]
@PatMulligan have you changed this setting before or after you created the refund? If you changed it AFTER you created the refund it will definitely not work as the old expiration time will stick. To get around this issue you will need to archive the pull payment, adjust the setting to 1 day and then create the refund again.

If that still doesn't work then it might not work at all unless we allow specifying <1 day expiration time for invoices/pull payments.

**Reply by PatMulligan** [reply]
yeah, I created a lightning refund on BTCPay setting it to 1 day (the lowest value accepted) and after that I asked my client to try claiming it with their muun generated invoice (fixed 24 hr invoice expiration), but it does not work.

**Reply by bolatovumar** [reply]
Tested on my instance of BTCPay and Muun wallet and got the same issue.

**Reply by bolatovumar** [reply]
Ok, I figured out what the problem is. Since it takes some time for the end user input their LN invoice into BTCPay input field the expiry of the invoice is now in less than one day so it doesn't pass. I think we need to allow creating pull payments which allow expiry time of less than one day as most wallets use 24 hours expiration time and don't allow you to change it.

@PatMulligan if you want to get around this issue for now you can ask the user to use Phoenix wallet or Simple Bitcoin Wallet which set the expiry time at 7 and 14 days respectively. See here: https://github.com/btcpayserver/btcpayserver/pull/3776#issuecomment-1140380681
